sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -467,6 +467,42 @@ class CudfRegexTranspiler(mode: RegexMode) {
     cudfRegex.toRegexString
   }
 
+  private def isSomeRepetition(f: ListBuffer[RegexAST] => Boolean)(e: RegexAST): Boolean = {
+    e match {
+      case RegexRepetition(_, _) => true
+      case RegexGroup(_, term) => isRepetition(term)
+      case RegexSequence(parts) if f(parts) => isRepetition(parts.last)
+      case _ => false
+    }
+  }
+
+  private def isRepetition = isSomeRepetition(_.nonEmpty)(_)
+  private def isNestedRepetition = isSomeRepetition(_.length == 1)(_)
+
+  private def isSupportedRepetitionBase(e: RegexAST): Boolean = {
+    e match {
+      case RegexEscaped(ch) if ch != 'd' && ch != 'w' => // example: "\B?"
+        false
+
+      case RegexChar(a) if "$^".contains(a) =>
+        // example: "$*"
+        false
+
+      case RegexRepetition(_, _) =>
+        // example: "a*+"
+        false
+
+      case RegexSequence(parts) =>
+        parts.forall(x => isSupportedRepetitionBase(x))
```

Comment:
nit: we should be able to simplify this

```suggestion
        parts.forall(isSupportedRepetitionBase)
```